### PR TITLE
bcm2835-bootloader: cleanup update.sh

### PR DESCRIPTION
**Pull request by antonlacon**
This changes bcm2835-bootloader to act more like u-boot projects in checking for update.sh and canupdate.sh among the project or device files.

It then cleans up update.sh to remove the leading indent, misplaced comment, and code commented for years.

### CODE REVIEW

File: packages/tools/bcm2835-bootloader/package.mk [hunk 1]
```diff
@@ -22,7 +22,17 @@ makeinstall_target() {
...
+      cp -av $PROJECT_DIR/$PROJECT/devices/$DEVICE/bootloader/update.sh $INSTALL/usr/share/bootloader/
+    elif [ -f $PROJECT_DIR/$PROJECT/bootloader/update.sh ]; then
+      cp -av $PROJECT_DIR/$PROJECT/bootloader/update.sh $INSTALL/usr/share/bootloader/
+    fi
+
+    if [ -f $PROJECT_DIR/$PROJECT/devices/$DEVICE/bootloader/canupdate.sh ]; then
+      cp -av $PROJECT_DIR/$PROJECT/devices/$DEVICE/bootloader/canupdate.sh $INSTALL/usr/share/bootloader/
+    elif [ -f $PROJECT_DIR/$PROJECT/bootloader/canupdate.sh ]; then
+      cp -av $PROJECT_DIR/$PROJECT/bootloader/canupdate.sh $INSTALL/usr/share/bootloader/
+    fi
```
Comment (MilhouseVH) [inline packages/tools/bcm2835-bootloader/package.mk]: Shouldn't we use `find_file_path bootloader/update.sh` and `find_file_path bootloader/canupdate.sh` (same as lines above & below)?

**Comment by vpeter4** [comment]
Is reason for `drop leading spaces` some new coding rule?

**Comment by antonlacon** [comment]
It's a change from the past, discussed in #3055. In short, having a comment isn't reason enough to indent code. I change it when I'm already making edits.

**Comment by vpeter4** [comment]
But you are violating a rule with those changes?

**Comment by antonlacon** [comment]
I'm aligning it to my understanding of:

> Initial entries have no indent. Something within an If-statement or for-loop gets indented. Comments are attached to the code and follow the indent of the code it's commenting on.

What rule do you believe this violates?

**Comment by vpeter4** [comment]
This one `Indent using 2 spaces within comments` https://github.com/LibreELEC/LibreELEC.tv/blob/1a1d8cf94bc5288eb6aa3b265f5633f3a6f424af/STANDARDS.md

Or I missread something?

File: packages/tools/bcm2835-bootloader/package.mk (hunk 1 above)
Comment (antonlacon) [reply packages/tools/bcm2835-bootloader/package.mk]: Now using find_file_path. Cheers

**Comment by antonlacon** [comment]
Your reading is correct, but I don't think that is done in practice any longer. I'll let @MilhouseVH weigh in when he gets a chance.

**Comment by MilhouseVH** [comment]
Ok, first of all the "indent using 2-spaces" rule applies to "build system" files (ie. `scripts/install`, `config/functions` etc.), and I'm not sure if I would class a script that runs on the target as a "build system" file - we should have an additional section for that.

For a start you can't assume that the script on the target will be `/bin/bash` (should be `/bin/sh`) so even the first "build system" rule is violated by any and every target script, and if the script is being added to help support a third-party package or process then it might even be best to use the third-party project coding standards.

So I think we need some simple guidelines for target scripts, probably a subset of the build system rules, adapted for `/bin/sh` etc.

The "indent using 2 spaces" rule itself has always been something of a quirk of the build system scripts and the current proposed coding standard may require updating as I posted [this](https://github.com/LibreELEC/LibreELEC.tv/pull/3055#issuecomment-433019003) proposal for comments to be indented the same as the commented code which received three up votes and no down votes, but for some reason that wording isn't yet reflected in the coding standard - will try and correct that later today.

tl;dr: @antonlacon is correct, and the RFC wording needs to be updated! :)

**Comment by vpeter4** [comment]
In my opinion all such indentations are ugly.
